zfs/zfs.go

```diff
@@ -1032,6 +1037,34 @@ type RecvOptions struct {
 	RollbackAndForceRecv bool
 	// Set -s flag used for resumable send & recv
 	SavePartialRecvState bool
+	InheritProperties    []Property
+	OverrideProperties   map[Property]string
+}
+
+type Property string
+
+func (p Property) Validate() error {
+	// see zfsprops(8), section "User Properties"
+	if len(p) < 1 {
+		return fmt.Errorf("property name can't be empty")
+	}
+	if len(p) > 256 {
```

Comment:
please find the ZFS constant, extract it to a const variable and put a comment over it. also make sure that C max string lengths often include the `\0` character